src/server.h

```diff
@@ -2354,6 +2359,23 @@ typedef struct {
     unsigned char *lpi; /* listpack iterator */
 } setTypeIterator;
 
+/* Structure for DB iterator that allows iterating across multiple slot specific dictionaries in cluster mode. */
+typedef struct dbIterator {
+    redisDb *db;
+    int index;
+    int cur_slot;
+} dbIterator;
```

Comment:
I think we could easily modify this so that the iterator can iterate both the keys and the dictionaries together, so there is one interface that can grab both.

Reply:
Okay, added `dictIterator` field to `dbIterator`, and introduced ability to iterate entries using new `dbIteratorNext` function.